### PR TITLE
feat: adapt camera field to support 2d mode

### PR DESCRIPTION
**Pull request by airslice**
close https://github.com/reearth/reearth/issues/285

# Overview

This task is aiming to update CameraField and related (like FovSlider) at UI level to support mode 2d.
Basiclly hiding or disable the pose parameters / UI. 

## What I've done

- add a `onlyPosition` parameter to `CameraField`
- get the sceneMode value from gql and create a status and passing it down to the `CameraField`
- hide the pose part of `CameraField` when use 2d mode
- disable `CameraField` when `onlyPose` and `onlyPosition` at the same time
- hide the `FovSlider` when use 2d mode

## How I tested

Switch scene mode and test CameraField and FovSlider

## Screenshot

image

image

## Which point I want you to review particularly

i meet the requirement that the property value inside the right panel (sceneProperty.default.sceneMode) will have effects on the other components (like CameraField for now) also inside the right panel (include the ones in other groups). The way i handle this is maybe not a good way i think.

### CODE REVIEW

File: src/components/organisms/EarthEditor/PropertyPane/hooks-queries.ts [hunk 1]
```diff
@@ -191,6 +191,12 @@ export default ({
 
   const layers = useMemo(() => convertLayers(layerData), [layerData]);
 
+  const sceneMode = useMemo(() => {
+    const defaultPropery = sceneProperty?.items.find(item => item.schemaGroupId === "default");
```
Comment (KaWaite) [inline src/components/organisms/EarthEditor/PropertyPane/hooks-queries.ts:195]: defaultProperty

**Comment by airslice** [comment]
updates:
- add a sceneMode status 
- add a setter inside the CanvasArea as it always get the scene property
- remove the property value finder for scene mode from the hooks-queryies of PropertyPane

File: src/components/organisms/EarthEditor/PropertyPane/hooks-queries.ts (hunk 1 above)
Comment (airslice) [reply src/components/organisms/EarthEditor/PropertyPane/hooks-queries.ts]: this part has been removed.

File: src/state/index.ts [hunk 2]
```diff
@@ -34,6 +34,10 @@ export const useIsCapturing = () => useAtom(isCapturing);
 const camera = atom<Camera | undefined>(undefined);
 export const useCamera = () => useAtom(camera);
 
+export type SceneMode = "3d" | "2d" | "columbus";
+const sceneMode = atom<SceneMode | undefined>("3d");
```
Comment (KaWaite) [inline src/state/index.ts:38]: Is this actually ever undefined? It starts as "3d" and I'm guessing there isn't (or shouldn't be) any way to set it to undefined, right? Or is this based on Cesium types?

**Review by KaWaite — APPROVED** [review]
Just one comment. If it is necessary the way it is, all looks fine!

File: src/state/index.ts (hunk 2 above)
Comment (airslice) [reply src/state/index.ts:38]: emmm yes there shouldn't be any way to set it to undefined. i'll update this part and the setter.

**Pull request merged**